rose/client/main.py

```diff
@@ -70,10 +72,8 @@ def main():
         print 'usage: rose-client drive-module'
         sys.exit(2)
 
-    with open(sys.argv[1]) as f:
-        d = {}
-        exec f in d, d
+    d = import_module(splitext(sys.argv[1])[0])
```

Comment:
Can you separate the module name extraction to separate line to make the code easier to follow?

For example:

    module_name = splitext(sys.argv[1])[0]

This code will work only when using:

    rose-client driver.py

But we have to support also importing from another directory:

    rose-client ../my-examples/driver.py

The import fails now:

```
$ ./rose-client examples/none.py 
Traceback (most recent call last):
  File "./rose-client", line 4, in <module>
    main.main()
  File "/home/nsoffer/src/rose/rose/client/main.py", line 75, in main
    d = import_module(splitext(sys.argv[1])[0])
  File "/usr/lib64/python2.7/importlib/__init__.py", line 37, in import_module
    __import__(name)
ImportError: Import by filename is not supported.
```

"d" was good name for a dictionary, but here we have a drive module, so maybe we should cal lit "mod"?

It would be nice to add a test for this code, probably extracting the driver loading code to a separate function (e.g. load_driver_module()), and testing that loading a driver file with relative and absolute path return the expected object.

Reply:
I updated the pull request.
unfortunately python doesn't support import a file without destroying the PYTHONPATH.
the only way to make this work is to make examples a python package with __init__.py which I don't think make sense.
Is running a driver file from a different folder a requirement?

Reply:
Yes, I usually run drivers from another repo like this:

    ./rose-client ../rose-dev/score.py

Modifying sys.path or requiring that the driver modules are inside a package is ok, but this tool is for young kids and I don't want to make it more complicated then needed. Providing a relative or absolute path to a file should work.

We use imp to implement this:
https://docs.python.org/2/library/imp.html

Reply:
Great suggestion!
Updated the pull request.
Also made sure the parameter is a python file (source or compiled).